File: boj_9267/solution.py
from math import gcd

def xGCD(a, b):
    if a < b:
        g, y, x = xGCD(b, a)
        return g, x, y
    if b == 0:
        return a, 1, 0
    g, x_, y_ = xGCD(b, a % b)
    return g, y_, x_ - (a // b) * y_


def main():
    a, b, S = map(int, input().split())
    if a == S or b == S or a + b == S:
        print('YES')
        return

    g = gcd(a, b)
    a //= g
    b //= g
    S //= g

    _, x, y = xGCD(a, b)
    x *= S
    y *= S

    if x < 0:
        k = (-x // b) + 1
        x += k * b
        y -= k * a
    if y < 0:
        k = (-y // b) + 1
        x -= k * b
        y += k * a

    # print(f'{x} * {a} + {y} * {b} => {x * a + y * b}, {S}')

    while y > 0:
        if x > 0 and y > 0 and gcd(x, y) == 1:
            print('YES')
            break
        x += b
        y -= a
    else:
        print('NO')

main()

File: boj_9267/solution.cpp
#include <bits/stdc++.h>
#include <cstdio>
using namespace std;

using LL = long long;

template <typename T = long long> // xgcd
T xGCD(T a, T b, T& x, T& y) {
  if (a < b) {
    return xGCD(b, a, y, x);
  }
  if (b == 0) {
    x = 1; y = 0;
    return a;
  }
  T x_, y_, g = xGCD(b, a % b, x_, y_);
  x = y_;
  y = x_ - (a / b) * y_;
  return g;
}

void solve() {
  LL a, b, S;
  scanf("%lld %lld %lld", &a, &b, &S);

  // trivial case
  if (a == S || b == S || a + b == S) {
    printf("YES\n");
    return;
  }

  LL x, y, g = xGCD(a, b, x, y);
  if (S % g != 0) {
    printf("NO\n");
    return;
  }
  LL n = S / g;
  x *= n;
  y *= n;
  while (x < 1) {
    x += b;
    y -= a;
  }
  //while (y < 1) {
    //x -= b;
    //y += a;
  //}
  printf("%lld * %lld + %lld * %lld = %lld\n", x, a, y, b, x * a + y * b);
  if (y < 1 || x < 1) {
    printf("NO\n");
  }
  else {
    printf("YES\n");
  }
}

int main() {
  int T = 1;
  while (T--) {
    solve();
  }

  return 0;
}
